pgdb/v1/fts.go

```diff
@@ -175,54 +275,58 @@ func acronymSplitDoc(docValue string, doc *SearchContent) []lexeme {
 		if unicode.IsUpper(prev) {
 			switch {
 			case unicode.IsLower(r):
-				// only append previous if it is upper case and and current is not lower case (i.e. don't append T in AWSTest)
-				if utf8.RuneCount(buffer.Bytes()) >= minWordSize {
-					rv = append(rv, lexeme{strings.ToLower(buffer.String()), pos, doc.Weight})
+				// only append previous if it is upper case and and current is not lower case (i.e. don't append T in AWSTest).
+				if utf8.RuneCount(wordBuffer.Bytes()) >= minWordSize {
+					rv = append(rv, lexeme{strings.ToLower(wordBuffer.String()), pos, doc.Weight})
 				}
-				buffer.Reset()
+				wordBuffer.Reset()
 			case !unicode.IsUpper(r):
-				// finish acronym if there is one of min length if we encounter space
-				if _, e := buffer.WriteRune(prev); e != nil {
-					buffer.Reset()
+				// finish acronym if there is one of min length if we encounter space.
+				if _, e := wordBuffer.WriteRune(prev); e != nil {
+					wordBuffer.Reset()
 					continue
 				}
-				if utf8.RuneCount(buffer.Bytes()) >= minWordSize {
-					rv = append(rv, lexeme{strings.ToLower(buffer.String()), pos, doc.Weight})
+				if utf8.RuneCount(wordBuffer.Bytes()) >= minWordSize {
+					rv = append(rv, lexeme{strings.ToLower(wordBuffer.String()), pos, doc.Weight})
 				}
-				buffer.Reset()
+				wordBuffer.Reset()
 			default:
-				if _, e := buffer.WriteRune(prev); e != nil {
-					buffer.Reset()
+				if _, e := wordBuffer.WriteRune(prev); e != nil {
+					wordBuffer.Reset()
 					continue
 				}
 			}
 		}
 		prev = r
 		pos += 1
 	}
-	// finish acronym if there is one of min length
-	if buffer.Len() > 0 {
+	// finish acronym if there is one of min length.
+	if wordBuffer.Len() > 0 {
 		if unicode.IsUpper(prev) {
-			if _, e := buffer.WriteRune(prev); e != nil {
+			if _, e := wordBuffer.WriteRune(prev); e != nil {
 				return rv
 			}
 		}
-		if utf8.RuneCount(buffer.Bytes()) >= minWordSize {
-			rv = append(rv, lexeme{strings.ToLower(buffer.String()), pos, doc.Weight})
+		if utf8.RuneCount(wordBuffer.Bytes()) >= minWordSize {
+			rv = append(rv, lexeme{strings.ToLower(wordBuffer.String()), pos, doc.Weight})
 		}
 	}
 	return rv
 }
 
-func camelSplitDocs(docs []*SearchContent) []lexeme {
+// normalizeVectorDocs - converts a set of input documents into a set of lexemes matching common patterns such as camel case, snake case and accronyms.
+func normalizeVectorDocs(docs []*SearchContent) []lexeme {
 	rv := make([]lexeme, 0, 8)
 	for _, doc := range docs {
 		if doc.Type == FieldOptions_FULL_TEXT_TYPE_ENGLISH_LONG {
 			continue
 		}
 		docValue := interfaceToValue(doc.Value)
-		rv = append(rv, camelSplitDoc(docValue, doc)...)
-		rv = append(rv, acronymSplitDoc(docValue, doc)...)
+		var wordBuffer bytes.Buffer
+		rv = append(rv, camelSplitDoc(docValue, wordBuffer, doc)...)
```

Comment:
is it possible to combine all of these?

right now we have to iterate the whole string multiple times....  4 times now?

Reply:
yes I prefer isolating the work these do, I don't think this will be noticeably slower to justify confusing them.